internal/expr/prepare.go

```diff
@@ -41,7 +82,91 @@ func prepareInput(ti typeNameToInfo, p *inputPart) error {
 	return nil
 }
 
-// Prepare takes a parsed expression and all the Go objects mentioned in it.
+// prepareOutput checks that the output expressions are correspond to a known types.
+// It then checks they are formatted correctly and finally generates the columns for the query.
+func prepareOutput(ti typeNameToInfo, p *outputPart) ([]fullName, error) {
+
+	var outCols []fullName = make([]fullName, 0)
+
+	// Check target struct type and its tags are valid.
+	var info *info
+	var ok bool
+
+	for i, t := range p.target {
+		if i == 0 {
+			info, ok = ti[t.prefix]
+			if !ok {
+				return nil, fmt.Errorf(`type %s unknown, have: %s`, t.prefix, strings.Join(getKeys(ti), ", "))
+			}
+		} else if t.prefix != info.structType.Name() {
+			return nil, fmt.Errorf("multiple target types in: %s", p.String())
```

Comment:
As we discussed, looks like we could handle this well as it's just a different spelling of the individual parts (`one AS &Foo.one, two AS &Bar.two` vs. `(one, two) AS (&Foo.one, &Bar.two)`).

Reply:
Changed, this is supported now.